# 🔎 Levantamento e Exploração das Fontes de Dados

Tech Challenge — Avaliação da Alfabetização

Este notebook faz parte da etapa de levantamento e exploração das fontes de dados utilizadas no desenvolvimento do projeto.

O objetivo desta etapa é conhecer a estrutura das bases disponíveis antes da implementação dos tratamentos e cruzamentos nas camadas da arquitetura de dados.

## 🎯 Objetivos deste levantamento

Nesta etapa buscamos:

- identificar as fontes de dados disponíveis para o projeto;
- identificar as tabelas, arquivos e estruturas existentes em cada fonte;
- levantar as colunas disponíveis em cada tabela;
- identificar os tipos de dados das variáveis;
- identificar possíveis chaves de relacionamento;
- verificar a cobertura temporal e geográfica das informações;
- identificar quais informações podem contribuir para responder às perguntas de negócio do projeto;
- avaliar a necessidade de fontes complementares.

## 🎯 Pergunta central do projeto

O levantamento das fontes será orientado pela necessidade de compreender o alcance das metas de alfabetização e investigar como os resultados podem se relacionar com características educacionais e socioeconômicas dos municípios e estados.

As perguntas de análise serão refinadas pela equipe à medida que as variáveis disponíveis nas diferentes fontes forem identificadas.

## 🏗️ Relação com a arquitetura do projeto

O levantamento realizado neste notebook antecede a implementação das camadas de dados na AWS.

O fluxo considerado para o projeto é:

**Fontes de dados → Bronze → Prata → Gold → Visualização**

Neste momento, o notebook **não tem como objetivo realizar a modelagem definitiva das camadas**.

A finalidade é entender o conteúdo e a estrutura das fontes para apoiar as decisões que serão tomadas posteriormente pela equipe.

### 🥉 Camada Bronze

A camada Bronze será responsável pelo armazenamento dos dados provenientes das fontes selecionadas, mantendo-os o mais próximo possível de sua origem.

O levantamento realizado neste notebook ajudará a equipe a definir quais arquivos, tabelas e informações deverão ser considerados nessa etapa.

### 🥈 Camada Prata

Na camada Prata serão realizados os tratamentos necessários sobre os dados, como:

- seleção das variáveis necessárias;
- padronização;
- tratamento de valores ausentes;
- ajustes de tipos;
- filtros;
- preparação dos dados para os relacionamentos.

### 🥇 Camada Gold

A camada Gold será construída posteriormente com os dados tratados e organizados para responder às perguntas definidas para o projeto.

Nesta etapa poderão ser criados indicadores, relacionamentos e estruturas específicas para análise e visualização.

## 📚 Fontes de dados analisadas

Para o desenvolvimento do projeto, foram exploradas três fontes de dados:

1. **Avaliação da Alfabetização — Base dos Dados**
   - Fonte principal do projeto.

2. **Censo Escolar 2023 — INEP**
   - Fonte complementar para informações sobre o contexto e a estrutura educacional.

3. **INSE 2023 — INEP**
   - Fonte complementar para informações relacionadas ao nível socioeconômico.

A seleção das fontes considerou a disponibilidade dos dados, o recorte temporal de 2023, a estrutura das informações e seu potencial de contribuição para as análises propostas.

## 📌 Metodologia do levantamento

Para cada fonte ou tabela analisada, são identificados:

1. nome da fonte;
2. nome da tabela ou arquivo;
3. quantidade de colunas;
4. nome das colunas;
5. tipo de dado;
6. possíveis chaves de relacionamento;
7. finalidade das informações dentro do projeto.

As consultas realizadas neste notebook têm caráter exploratório e documental. O objetivo é registrar o processo de descoberta e avaliação das fontes antes da implementação da arquitetura final.

## 1. Base Principal — Avaliação da Alfabetização



In [ ]:
!pip install -q basedosdados

   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 89.9/89.9 kB 1.9 MB/s eta 0:00:00
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 61.6/61.6 kB 3.4 MB/s eta 0:00:00
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 3.4/3.4 MB 17.3 MB/s eta 0:00:00
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 214.9/214.9 kB 9.0 MB/s eta 0:00:00


In [ ]:
import basedosdados as bd
from google.cloud import bigquery

### 1.1 Estrutura das tabelas

In [ ]:
PROJECT_ID = "primeval-melody-504216-n5"

client = bigquery.Client(project=PROJECT_ID)

print("Projeto configurado:", client.project)

Projeto configurado: primeval-melody-504216-n5


In [ ]:
from google.colab import auth

auth.authenticate_user()

### 1.2 Tabela `alunos`

In [ ]:
# TABELA ALUNOS
table_ref = "basedosdados.br_inep_avaliacao_alfabetizacao.alunos"

table = client.get_table(table_ref)

print("Tabela:", table.full_table_id)
print("Número de colunas:", len(table.schema))

Tabela: basedosdados:br_inep_avaliacao_alfabetizacao.alunos
Número de colunas: 12


In [ ]:
for campo in table.schema:
    print(f"{campo.name} → {campo.field_type}")

ano → INTEGER
id_municipio → STRING
id_escola → STRING
id_aluno → STRING
caderno → STRING
serie → STRING
rede → STRING
presenca → STRING
preenchimento_caderno → STRING
alfabetizado → STRING
proficiencia → FLOAT
peso_aluno → FLOAT


###1.3 Tabela Municipio

In [ ]:
# TABELA MUNICÍPIO

table_ref = "basedosdados.br_inep_avaliacao_alfabetizacao.municipio"

In [ ]:
table = client.get_table(table_ref)

print("Tabela:", table.full_table_id)
print("Número de colunas:", len(table.schema))

Tabela: basedosdados:br_inep_avaliacao_alfabetizacao.municipio
Número de colunas: 15


In [ ]:
for campo in table.schema:
    print(f"{campo.name} → {campo.field_type}")

ano → INTEGER
id_municipio → STRING
serie → STRING
rede → STRING
taxa_alfabetizacao → FLOAT
media_portugues → FLOAT
proporcao_aluno_nivel_0 → FLOAT
proporcao_aluno_nivel_1 → FLOAT
proporcao_aluno_nivel_2 → FLOAT
proporcao_aluno_nivel_3 → FLOAT
proporcao_aluno_nivel_4 → FLOAT
proporcao_aluno_nivel_5 → FLOAT
proporcao_aluno_nivel_6 → FLOAT
proporcao_aluno_nivel_7 → FLOAT
proporcao_aluno_nivel_8 → FLOAT


Observação

A tabela `municipio` apresenta indicadores agregados por município,
incluindo taxa de alfabetização, média de Português e distribuição
dos alunos segundo níveis de proficiência.

O campo `id_municipio` será posteriormente avaliado como possível
chave de relacionamento com as demais bases utilizadas no projeto.

### 1.4 Tabela `uf`

In [ ]:
# TABELA UF

table_ref = "basedosdados.br_inep_avaliacao_alfabetizacao.uf"

table = client.get_table(table_ref)

print("Tabela:", table.full_table_id)
print("Número de colunas:", len(table.schema))

for campo in table.schema:
    print(f"{campo.name} → {campo.field_type}")

Tabela: basedosdados:br_inep_avaliacao_alfabetizacao.uf
Número de colunas: 15
ano → INTEGER
sigla_uf → STRING
serie → STRING
rede → STRING
taxa_alfabetizacao → FLOAT
media_portugues → FLOAT
proporcao_aluno_nivel_0 → FLOAT
proporcao_aluno_nivel_1 → FLOAT
proporcao_aluno_nivel_2 → FLOAT
proporcao_aluno_nivel_3 → FLOAT
proporcao_aluno_nivel_4 → FLOAT
proporcao_aluno_nivel_5 → FLOAT
proporcao_aluno_nivel_6 → FLOAT
proporcao_aluno_nivel_7 → FLOAT
proporcao_aluno_nivel_8 → FLOAT


### 1.5 Tabela `meta_alfabetizacao_brasil`

In [ ]:
# TABELA ALFABETIZAÇÃO BRASIL

table_ref = "basedosdados.br_inep_avaliacao_alfabetizacao.meta_alfabetizacao_brasil"

table = client.get_table(table_ref)

print("Tabela:", table.full_table_id)
print("Número de colunas:", len(table.schema))

Tabela: basedosdados:br_inep_avaliacao_alfabetizacao.meta_alfabetizacao_brasil
Número de colunas: 11


In [ ]:
for campo in table.schema:
    print(f"{campo.name} → {campo.field_type}")

ano → INTEGER
rede → STRING
taxa_alfabetizacao → FLOAT
meta_alfabetizacao_2024 → FLOAT
meta_alfabetizacao_2025 → FLOAT
meta_alfabetizacao_2026 → FLOAT
meta_alfabetizacao_2027 → FLOAT
meta_alfabetizacao_2028 → FLOAT
meta_alfabetizacao_2029 → FLOAT
meta_alfabetizacao_2030 → FLOAT
percentual_participacao → FLOAT


Observação

A tabela `meta_alfabetizacao_brasil` possui 11 colunas e apresenta
informações agregadas em nível nacional.

Além do ano, rede de ensino e taxa de alfabetização, a tabela
disponibiliza metas de alfabetização projetadas para o período de
2024 a 2030 e o percentual de participação.

A presença da taxa de alfabetização e das metas em uma mesma tabela
permite, posteriormente, avaliar a distância entre o resultado
observado e as metas estabelecidas.

### 1.6 Tabela `meta_alfabetizacao_municipio`

In [ ]:
# TABELA ALFABETIZACAO MUNICIPIO

table_ref = "basedosdados.br_inep_avaliacao_alfabetizacao.meta_alfabetizacao_municipio"

table = client.get_table(table_ref)

print("Tabela:", table.full_table_id)
print("Número de colunas:", len(table.schema))

Tabela: basedosdados:br_inep_avaliacao_alfabetizacao.meta_alfabetizacao_municipio
Número de colunas: 13


In [ ]:
for campo in table.schema:
    print(f"{campo.name} → {campo.field_type}")

ano → INTEGER
id_municipio → STRING
rede → STRING
taxa_alfabetizacao → FLOAT
meta_alfabetizacao_2024 → FLOAT
meta_alfabetizacao_2025 → FLOAT
meta_alfabetizacao_2026 → FLOAT
meta_alfabetizacao_2027 → FLOAT
meta_alfabetizacao_2028 → FLOAT
meta_alfabetizacao_2029 → FLOAT
meta_alfabetizacao_2030 → FLOAT
nivel_alfabetizacao → INTEGER
percentual_participacao → FLOAT


Observação

A tabela `meta_alfabetizacao_municipio` possui 13 colunas e apresenta
informações agregadas em nível municipal.

Além da taxa e do nível de alfabetização, a tabela disponibiliza
metas projetadas para o período de 2024 a 2030, percentual de
participação e o identificador `id_municipio`.

O campo `id_municipio` será posteriormente avaliado como chave de
relacionamento com outras tabelas da base de Avaliação da
Alfabetização e com bases educacionais adicionais que possuam o
mesmo identificador ou mecanismo de correspondência municipal.

### 1.7 Tabela `meta_alfabetizacao_uf`

In [ ]:
# TABELA ALFABETIZACAO UF

table_ref = "basedosdados.br_inep_avaliacao_alfabetizacao.meta_alfabetizacao_uf"

table = client.get_table(table_ref)

print("Tabela:", table.full_table_id)
print("Número de colunas:", len(table.schema))

Tabela: basedosdados:br_inep_avaliacao_alfabetizacao.meta_alfabetizacao_uf
Número de colunas: 12


In [ ]:
for campo in table.schema:
    print(f"{campo.name} → {campo.field_type}")

ano → INTEGER
sigla_uf → STRING
rede → STRING
taxa_alfabetizacao → FLOAT
meta_alfabetizacao_2024 → FLOAT
meta_alfabetizacao_2025 → FLOAT
meta_alfabetizacao_2026 → FLOAT
meta_alfabetizacao_2027 → FLOAT
meta_alfabetizacao_2028 → FLOAT
meta_alfabetizacao_2029 → FLOAT
meta_alfabetizacao_2030 → FLOAT
percentual_participacao → FLOAT


### 1.8 Tabela `dicionario`

In [ ]:
table_ref = "basedosdados.br_inep_avaliacao_alfabetizacao.dicionario"

table = client.get_table(table_ref)

print("Tabela:", table.full_table_id)
print("Número de colunas:", len(table.schema))

Tabela: basedosdados:br_inep_avaliacao_alfabetizacao.dicionario
Número de colunas: 5


In [ ]:
for campo in table.schema:
    print(f"{campo.name} → {campo.field_type}")

id_tabela → STRING
nome_coluna → STRING
chave → STRING
cobertura_temporal → STRING
valor → STRING


Observação

A tabela `dicionario` possui 5 colunas e funciona como uma tabela
de metadados da base.

Os campos `id_tabela` e `nome_coluna` permitem identificar a qual
tabela e variável determinada informação está relacionada. Os
campos `chave`, `cobertura_temporal` e `valor` complementam as
informações necessárias para interpretação das variáveis.

Por se tratar de uma tabela de apoio, seus dados não representam
diretamente os resultados de alfabetização, mas serão utilizados
como referência para interpretação e documentação das demais
tabelas.

### 1.9 🎯 Objetivos da análise — Metas de Alfabetização

A Avaliação da Alfabetização é a base principal deste projeto e tem
como objetivo analisar os resultados de alfabetização e sua relação
com as metas estabelecidas.

Nesta etapa, serão analisadas as tabelas relacionadas às metas em
diferentes níveis geográficos, permitindo posteriormente realizar
cruzamentos com outras bases educacionais.

**Estrutura das tabelas de metas**

| Nível | Tabela | Objetivo da análise |
|---|---|---|
| 🇧🇷 Brasil | `meta_alfabetizacao_brasil` | Identificar a meta nacional de alfabetização |
| 🏙️ Município | `meta_alfabetizacao_municipio` | Analisar as metas estabelecidas para cada município |
| 🗺️ UF | `meta_alfabetizacao_uf` | Comparar as metas e resultados entre as Unidades da Federação |
| 📍 Referência | `uf` | Identificar e relacionar as Unidades da Federação |

**🔎 Perguntas que pretendemos responder**

A análise será orientada pelas seguintes perguntas:

### 1. Alcance das metas
- Qual é a meta de alfabetização estabelecida para o período analisado?
- Quais municípios atingiram ou superaram a meta?
- Quais municípios ficaram abaixo da meta?
- Qual é a proporção de municípios que atingiram a meta?

### 2. Comparação entre estados
- Quais Unidades da Federação apresentam os melhores resultados?
- Quais estados apresentam maior distância em relação à meta?
- Existem diferenças relevantes no desempenho entre as UFs?
- Quais estados apresentam maior concentração de municípios abaixo da meta?

### 3. Desempenho e alfabetização
- Existe relação entre a taxa de alfabetização e o alcance da meta?
- Como a distribuição dos níveis de proficiência se relaciona com o
  cumprimento das metas?
- Quais municípios apresentam bons resultados mesmo estando abaixo
  da meta?

### 4. Cruzamento com outras bases
Posteriormente, os resultados das metas poderão ser relacionados a outras bases selecionadas, como:

- **Censo Escolar:** infraestrutura, matrículas, docentes e características das escolas;
- **INSE:** nível socioeconômico dos estudantes e das escolas.

Esses cruzamentos permitirão investigar possíveis relações entre o
contexto educacional e socioeconômico e o alcance das metas de
alfabetização.

**🎯 Resultado esperado**

A partir dos cruzamentos, pretende-se construir indicadores que
permitam visualizar:

**Meta → Resultado → Diferença → Contexto educacional → Contexto socioeconômico**

Esses indicadores serão utilizados posteriormente na definição da
**camada GOLD** e na construção da visualização.

## Inventário da Base Principal

## Base: `br_inep_avaliacao_alfabetizacao`

A base de Avaliação da Alfabetização é a base principal do projeto.
Abaixo está o inventário das tabelas identificadas, seus campos e
respectivos tipos de dados.

---

## 1. Tabela `alunos`

**Quantidade de colunas:** 12

| Campo | Tipo |
|---|---|
| `ano` | INTEGER |
| `id_municipio` | STRING |
| `id_escola` | STRING |
| `id_aluno` | STRING |
| `caderno` | STRING |
| `serie` | STRING |
| `rede` | STRING |
| `presenca` | STRING |
| `preenchimento_caderno` | STRING |
| `alfabetizado` | STRING |
| `proficiencia` | FLOAT |
| `peso_aluno` | FLOAT |

**Finalidade:** dados relacionados aos alunos participantes da
avaliação, incluindo informações de identificação, participação,
alfabetização e proficiência.

---

## 2. Tabela `municipio`

**Quantidade de colunas:** 15

| Campo | Tipo |
|---|---|
| `ano` | INTEGER |
| `id_municipio` | STRING |
| `serie` | STRING |
| `rede` | STRING |
| `taxa_alfabetizacao` | FLOAT |
| `media_portugues` | FLOAT |
| `proporcao_aluno_nivel_0` | FLOAT |
| `proporcao_aluno_nivel_1` | FLOAT |
| `proporcao_aluno_nivel_2` | FLOAT |
| `proporcao_aluno_nivel_3` | FLOAT |
| `proporcao_aluno_nivel_4` | FLOAT |
| `proporcao_aluno_nivel_5` | FLOAT |
| `proporcao_aluno_nivel_6` | FLOAT |
| `proporcao_aluno_nivel_7` | FLOAT |
| `proporcao_aluno_nivel_8` | FLOAT |

**Finalidade:** indicadores agregados de alfabetização e desempenho
em nível municipal.

---

## 3. Tabela `uf`

**Quantidade de colunas:** 15

| Campo | Tipo |
|---|---|
| `ano` | INTEGER |
| `sigla_uf` | STRING |
| `serie` | STRING |
| `rede` | STRING |
| `taxa_alfabetizacao` | FLOAT |
| `media_portugues` | FLOAT |
| `proporcao_aluno_nivel_0` | FLOAT |
| `proporcao_aluno_nivel_1` | FLOAT |
| `proporcao_aluno_nivel_2` | FLOAT |
| `proporcao_aluno_nivel_3` | FLOAT |
| `proporcao_aluno_nivel_4` | FLOAT |
| `proporcao_aluno_nivel_5` | FLOAT |
| `proporcao_aluno_nivel_6` | FLOAT |
| `proporcao_aluno_nivel_7` | FLOAT |
| `proporcao_aluno_nivel_8` | FLOAT |

**Finalidade:** indicadores agregados de alfabetização e desempenho
em nível estadual.

---

## 4. Tabela `meta_alfabetizacao_brasil`

**Quantidade de colunas:** 11

| Campo | Tipo |
|---|---|
| `ano` | INTEGER |
| `rede` | STRING |
| `taxa_alfabetizacao` | FLOAT |
| `meta_alfabetizacao_2024` | FLOAT |
| `meta_alfabetizacao_2025` | FLOAT |
| `meta_alfabetizacao_2026` | FLOAT |
| `meta_alfabetizacao_2027` | FLOAT |
| `meta_alfabetizacao_2028` | FLOAT |
| `meta_alfabetizacao_2029` | FLOAT |
| `meta_alfabetizacao_2030` | FLOAT |
| `percentual_participacao` | FLOAT |

**Finalidade:** metas e indicadores de alfabetização em nível nacional.

---

## 5. Tabela `meta_alfabetizacao_municipio`

**Quantidade de colunas:** 13

| Campo | Tipo |
|---|---|
| `ano` | INTEGER |
| `id_municipio` | STRING |
| `rede` | STRING |
| `taxa_alfabetizacao` | FLOAT |
| `meta_alfabetizacao_2024` | FLOAT |
| `meta_alfabetizacao_2025` | FLOAT |
| `meta_alfabetizacao_2026` | FLOAT |
| `meta_alfabetizacao_2027` | FLOAT |
| `meta_alfabetizacao_2028` | FLOAT |
| `meta_alfabetizacao_2029` | FLOAT |
| `meta_alfabetizacao_2030` | FLOAT |
| `nivel_alfabetizacao` | INTEGER |
| `percentual_participacao` | FLOAT |

**Finalidade:** metas e indicadores de alfabetização em nível municipal.

---

## 6. Tabela `meta_alfabetizacao_uf`

**Quantidade de colunas:** 12

| Campo | Tipo |
|---|---|
| `ano` | INTEGER |
| `sigla_uf` | STRING |
| `rede` | STRING |
| `taxa_alfabetizacao` | FLOAT |
| `meta_alfabetizacao_2024` | FLOAT |
| `meta_alfabetizacao_2025` | FLOAT |
| `meta_alfabetizacao_2026` | FLOAT |
| `meta_alfabetizacao_2027` | FLOAT |
| `meta_alfabetizacao_2028` | FLOAT |
| `meta_alfabetizacao_2029` | FLOAT |
| `meta_alfabetizacao_2030` | FLOAT |
| `percentual_participacao` | FLOAT |

**Finalidade:** metas e indicadores de alfabetização em nível estadual.

---

## 7. Tabela `dicionario`

**Quantidade de colunas:** 5

| Campo | Tipo |
|---|---|
| `id_tabela` | STRING |
| `nome_coluna` | STRING |
| `chave` | STRING |
| `cobertura_temporal` | STRING |
| `valor` | STRING |

**Finalidade:** tabela de metadados utilizada como apoio à
interpretação das variáveis das demais tabelas.

---

# 🔑 Principais campos para relacionamento

| Campo | Tabelas | Possível utilização |
|---|---|---|
| `id_municipio` | `alunos`, `municipio`, `meta_alfabetizacao_municipio` | Relacionamento em nível municipal |
| `sigla_uf` | `uf`, `meta_alfabetizacao_uf` | Relacionamento em nível estadual |
| `ano` | diversas tabelas | Relacionamento temporal |
| `rede` | diversas tabelas | Segmentação por rede de ensino |
| `serie` | `alunos`, `municipio`, `uf` | Segmentação por série |

## Consulta de municípios - API IBGE

Para complementar as bases utilizadas no projeto, será utilizada a
API de Localidades do IBGE para obter a relação de municípios
brasileiros.

A API disponibiliza o código IBGE do município e seu respectivo nome,
permitindo relacionar essas informações ao campo `id_municipio`
presente nas tabelas da Base dos Dados.

Endpoint utilizado:

https://servicodados.ibge.gov.br/api/v1/localidades/municipios

A consulta não requer autenticação.

In [ ]:
import requests
import pandas as pd

url = "https://servicodados.ibge.gov.br/api/v1/localidades/municipios"

resposta = requests.get(url)
resposta.raise_for_status()

dados = resposta.json()

print(f"Quantidade de registros retornados: {len(dados)}")

Quantidade de registros retornados: 5571


In [ ]:
municipios = pd.DataFrame([
    {
        "id_municipio": str(municipio["id"]),
        "nome_municipio": municipio["nome"]
    }
    for municipio in dados
])

municipios.head()

,id_municipio,nome_municipio
0,1100015,Alta Floresta D'Oeste
1,1100023,Ariquemes
2,1100031,Cabixi
3,1100049,Cacoal
4,1100056,Cerejeiras


In [ ]:
print("Quantidade de municípios:", len(municipios))
print("Códigos de municípios únicos:", municipios["id_municipio"].nunique())

Quantidade de municípios: 5571
Códigos de municípios únicos: 5571


In [ ]:
print("IDs nulos:", municipios["id_municipio"].isna().sum())
print("Nomes nulos:", municipios["nome_municipio"].isna().sum())

IDs nulos: 0
Nomes nulos: 0


Observação

A API de Localidades do IBGE foi consultada para verificar a
disponibilidade de uma fonte oficial contendo o código e o nome
dos municípios brasileiros.

A consulta retornou 5.571 municípios, com 5.571 códigos únicos e
sem valores nulos para código ou nome do município.

Nesta etapa, os dados são utilizados exclusivamente para
levantamento e avaliação da fonte. A definição das colunas que
serão carregadas para a camada Bronze e os tratamentos que serão
realizados na camada Prata serão definidos posteriormente,
conforme a modelagem do projeto.

# 🏫 Fonte Complementar 1 — Censo Escolar 2023

O Censo Escolar será analisado como fonte complementar à Avaliação
da Alfabetização, com o objetivo de identificar informações
relacionadas à estrutura e ao contexto da educação básica.

Nesta etapa será realizado um levantamento completo da estrutura dos
dados disponibilizados pelo INEP, sem aplicação de filtros ou
seleção prévia de variáveis.

Serão identificados:

- arquivos e estruturas disponíveis;
- tabelas/estruturas de dados;
- quantidade de colunas;
- nomes das colunas;
- tipos de dados;
- possíveis chaves de relacionamento;
- informações relevantes para análises posteriores.

A definição das variáveis que serão utilizadas nas camadas Bronze,
Prata e Gold será realizada posteriormente pela equipe.

## 🔎 2.1 Estrutura: microdados_unidade_coleta

A estrutura `microdados_unidade_coleta` representa os dados dos
microdados da unidade de coleta do Censo Escolar da Educação Básica
2023.

Nesta etapa será realizado o levantamento das variáveis disponíveis,
sem aplicação de filtros.

## 📖 2.2 Dicionário de Dados — Censo Escolar 2023

O dicionário de dados disponibilizado pelo INEP será utilizado como
referência para identificar as estruturas e variáveis existentes
nos microdados do Censo Escolar.

In [ ]:
from google.colab import files

arquivo = files.upload()


Saving dicionário_dados_educação_básica.xlsx to dicionário_dados_educação_básica.xlsx


In [ ]:
import os

print(os.listdir())


['.config', 'dicionário_dados_educação_básica.xlsx', 'sample_data']


In [ ]:
import pandas as pd

arquivo_dicionario = "dicionário_dados_educação_básica.xlsx"

excel = pd.ExcelFile(arquivo_dicionario)

print("Abas disponíveis:")
for aba in excel.sheet_names:
    print(f"- {aba}")

Abas disponíveis:
- microdados_unidade_coleta
- suplemento_cursos_tecnicos


In [ ]:
df_dicionario = pd.read_excel(
    arquivo_dicionario,
    sheet_name="microdados_unidade_coleta"
)

print("Linhas:", df_dicionario.shape[0])
print("Colunas:", df_dicionario.shape[1])

print("\nNome das colunas:")
for coluna in df_dicionario.columns:
    print(f"- {coluna}")

Linhas: 471
Colunas: 24

Nome das colunas:
- Unnamed: 0
- Alteração de nomenclatura ou metodologia de cálculo
- Unnamed: 2
- Unnamed: 3
- Unnamed: 4
- Unnamed: 5
- Unnamed: 6
- Unnamed: 7
- Unnamed: 8
- Unnamed: 9
- Unnamed: 10
- Unnamed: 11
- Unnamed: 12
- Unnamed: 13
- Unnamed: 14
- Unnamed: 15
- Unnamed: 16
- Unnamed: 17
- Unnamed: 18
- Unnamed: 19
- Unnamed: 20
- Unnamed: 21
- Unnamed: 22
- Unnamed: 23


Como nao localizamos o cabecarios na primeira linha vamos verificar as 15 primeiras linhas para identificar os dados

In [ ]:
df_bruto = pd.read_excel(
    arquivo_dicionario,
    sheet_name="microdados_unidade_coleta",
    header=None
)

df_bruto.head(15)


,0,1,2,3,4,5,6,7,8,9,...,14,15,16,17,18,19,20,21,22,23
0,NaN,Alteração de nomenclatura ou metodologia de cá...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
1,NaN,Variável nova,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
2,NaN,Descontinuidade,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
3,Dicionário de Variáveis - Tabela de Escolas,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
4,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
5,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
6,N,Nome da Variável,Descrição da Variável,Tipo,Tam.(1),Categoria,"Coleta por ano (""s""=sim;""n""=não )",NaN,NaN,NaN,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,Notas Importantes
7,NaN,NaN,NaN,NaN,NaN,NaN,07,08,09,10,...,15,16,17,18,19,20,21,22,23,NaN
8,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
9,1,NU_ANO_CENSO,Ano do Censo,Num,4,NaN,s,s,s,s,...,s,s,s,s,s,s,s,s,s,NaN


In [ ]:
df_censo = pd.read_excel(
    arquivo_dicionario,
    sheet_name="microdados_unidade_coleta",
    header=6
)

print("Linhas:", df_censo.shape[0])
print("Colunas:", df_censo.shape[1])

print("\nColunas do dicionário:")
for coluna in df_censo.columns:
    print(f"- {coluna}")

Linhas: 465
Colunas: 24

Colunas do dicionário:
- N
- Nome da Variável
- Descrição da Variável
- Tipo
- Tam.(1)
- Categoria
- Coleta por ano ("s"=sim;"n"=não ) 
- Unnamed: 7
- Unnamed: 8
- Unnamed: 9
- Unnamed: 10
- Unnamed: 11
- Unnamed: 12
- Unnamed: 13
- Unnamed: 14
- Unnamed: 15
- Unnamed: 16
- Unnamed: 17
- Unnamed: 18
- Unnamed: 19
- Unnamed: 20
- Unnamed: 21
- Unnamed: 22
- Notas Importantes


In [ ]:
# Identifica os anos presentes nas colunas de coleta
anos = df_censo.iloc[0, 7:24].tolist()

print("Anos de coleta:")
print(anos)

Anos de coleta:
['08', '09', '10', '11', '12', '13', '14', '15', '16', '17', '18', '19', '20', '21', '22', '23', nan]


In [ ]:
# Identifica somente os anos de coleta
anos = df_censo.iloc[0, 7:23].tolist()

print("Anos de coleta:")
print(anos)

Anos de coleta:
['08', '09', '10', '11', '12', '13', '14', '15', '16', '17', '18', '19', '20', '21', '22', '23']


In [ ]:
# Monta os nomes corretos das 24 colunas
novos_nomes = (
    list(df_censo.columns[:7])
    + anos
    + ["Notas Importantes"]
)

df_censo.columns = novos_nomes

print("Quantidade de colunas:", len(df_censo.columns))

print("\nColunas:")
for coluna in df_censo.columns:
    print(f"- {coluna}")

Quantidade de colunas: 24

Colunas:
- N
- Nome da Variável
- Descrição da Variável
- Tipo
- Tam.(1)
- Categoria
- Coleta por ano ("s"=sim;"n"=não ) 
- 08
- 09
- 10
- 11
- 12
- 13
- 14
- 15
- 16
- 17
- 18
- 19
- 20
- 21
- 22
- 23
- Notas Importantes


In [ ]:
df_censo.head(10)

,N,Nome da Variável,Descrição da Variável,Tipo,Tam.(1),Categoria,"Coleta por ano (""s""=sim;""n""=não )",08,09,10,...,15,16,17,18,19,20,21,22,23,Notas Importantes
0,NaN,NaN,NaN,NaN,NaN,NaN,07,08,09,10,...,15,16,17,18,19,20,21,22,23,NaN
1,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
2,1,NU_ANO_CENSO,Ano do Censo,Num,4.0,NaN,s,s,s,s,...,s,s,s,s,s,s,s,s,s,NaN
3,2,NO_REGIAO,Nome da Região Geográfica,Char,20.0,NaN,s,s,s,s,...,s,s,s,s,s,s,s,s,s,NaN
4,3,CO_REGIAO,Código da Região Geográfica,Num,1.0,NaN,s,s,s,s,...,s,s,s,s,s,s,s,s,s,NaN
5,4,NO_UF,Nome da Unidade da Federação,Char,50.0,NaN,s,s,s,s,...,s,s,s,s,s,s,s,s,s,NaN
6,5,SG_UF,Sigla da Unidade da Federação,Char,2.0,NaN,s,s,s,s,...,s,s,s,s,s,s,s,s,s,NaN
7,6,CO_UF,Código da Unidade da Federação,Num,2.0,NaN,s,s,s,s,...,s,s,s,s,s,s,s,s,s,NaN
8,7,NO_MUNICIPIO,Nome do Município,Char,150.0,NaN,s,s,s,s,...,s,s,s,s,s,s,s,s,s,NaN
9,8,CO_MUNICIPIO,Código do Município,Num,7.0,NaN,s,s,s,s,...,s,s,s,s,s,s,s,s,s,NaN


In [ ]:
inventario_censo = df_censo[
    ["N", "Nome da Variável", "Descrição da Variável", "Tipo", "Tam.(1)", "Categoria"]
].copy()

print("Quantidade de variáveis:", len(inventario_censo))

inventario_censo.head(20)

Quantidade de variáveis: 465


,N,Nome da Variável,Descrição da Variável,Tipo,Tam.(1),Categoria
0,NaN,NaN,NaN,NaN,NaN,NaN
1,NaN,NaN,NaN,NaN,NaN,NaN
2,1,NU_ANO_CENSO,Ano do Censo,Num,4.0,NaN
3,2,NO_REGIAO,Nome da Região Geográfica,Char,20.0,NaN
4,3,CO_REGIAO,Código da Região Geográfica,Num,1.0,NaN
5,4,NO_UF,Nome da Unidade da Federação,Char,50.0,NaN
6,5,SG_UF,Sigla da Unidade da Federação,Char,2.0,NaN
7,6,CO_UF,Código da Unidade da Federação,Num,2.0,NaN
8,7,NO_MUNICIPIO,Nome do Município,Char,150.0,NaN
9,8,CO_MUNICIPIO,Código do Município,Num,7.0,NaN


In [ ]:
inventario_censo = inventario_censo[
    inventario_censo["N"].notna()
].copy()

print("Quantidade de variáveis:", len(inventario_censo))

Quantidade de variáveis: 462


In [ ]:
linhas_sem_n = df_censo[df_censo["N"].isna()]

print("Linhas sem número de variável:", len(linhas_sem_n))

linhas_sem_n

Linhas sem número de variável: 3


,N,Nome da Variável,Descrição da Variável,Tipo,Tam.(1),Categoria,"Coleta por ano (""s""=sim;""n""=não )",08,09,10,...,15,16,17,18,19,20,21,22,23,Notas Importantes
0,NaN,NaN,NaN,NaN,NaN,NaN,07,08,09,10,...,15,16,17,18,19,20,21,22,23,NaN
1,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
460,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN


Observação sobre a estrutura do dicionário

A aba `microdados_unidade_coleta` possui 465 linhas e 24 colunas no arquivo original.

Durante a leitura, foram identificadas 3 linhas sem número de variável (`N`), utilizadas como elementos estruturais do próprio dicionário.

Considerando apenas os registros que possuem o campo `N` preenchido, foram identificadas **462 variáveis**.

As colunas de anos representam a disponibilidade da variável para coleta entre 2008 e 2023.

In [ ]:
# Inventário completo das variáveis disponíveis no Censo Escolar

print("Total de variáveis:", len(inventario_censo))

display(inventario_censo)

Total de variáveis: 462


,N,Nome da Variável,Descrição da Variável,Tipo,Tam.(1),Categoria
2,1,NU_ANO_CENSO,Ano do Censo,Num,4.0,NaN
3,2,NO_REGIAO,Nome da Região Geográfica,Char,20.0,NaN
4,3,CO_REGIAO,Código da Região Geográfica,Num,1.0,NaN
5,4,NO_UF,Nome da Unidade da Federação,Char,50.0,NaN
6,5,SG_UF,Sigla da Unidade da Federação,Char,2.0,NaN
...,...,...,...,...,...,...
459,458,QT_TUR_MED_INT,Número de Turmas do Ensino Médio - Tempo Integral,Num,8.0,NaN
461,Notas:,NaN,NaN,NaN,NaN,NaN
462,1) Tamanho do campo em termos de caracteres oc...,NaN,NaN,NaN,NaN,NaN
463,"2) De 2020 a 2023, os limites adotados para as...",NaN,NaN,NaN,NaN,NaN


In [ ]:
print("Tipos de variáveis:")
print(inventario_censo["Tipo"].value_counts())

Tipos de variáveis:
Tipo
Num     439
Char     17
Data      2
Name: count, dtype: int64


In [ ]:
print("\nCategorias preenchidas:")
print(inventario_censo["Categoria"].notna().sum())


Categorias preenchidas:
284


In [ ]:
tipos_diferentes = inventario_censo[
    ~inventario_censo["Tipo"].isin(["Num", "Char", "Data"])
]

print("Variáveis com tipo diferente ou não informado:", len(tipos_diferentes))

display(
    tipos_diferentes[
        ["N", "Nome da Variável", "Descrição da Variável", "Tipo"]
    ]
)

Variáveis com tipo diferente ou não informado: 4


,N,Nome da Variável,Descrição da Variável,Tipo
461,Notas:,NaN,NaN,NaN
462,1) Tamanho do campo em termos de caracteres oc...,NaN,NaN,NaN
463,"2) De 2020 a 2023, os limites adotados para as...",NaN,NaN,NaN
464,"3) Em 2019, os limites adotados para as variáv...",NaN,NaN,NaN


In [ ]:
inventario_censo = inventario_censo[
    inventario_censo["N"].apply(lambda x: str(x).isdigit())
].copy()

print("Quantidade de variáveis:", len(inventario_censo))

Quantidade de variáveis: 458


In [ ]:
print(inventario_censo["Tipo"].value_counts())

Tipo
Num     439
Char     17
Data      2
Name: count, dtype: int64


**Observação**

O dicionário da aba `microdados_unidade_coleta` apresenta 465 linhas e 24 colunas no arquivo original.

Após considerar apenas os registros com o campo `N` preenchido, foram identificados 462 registros. Desses, 4 correspondem a informações estruturais ou notas do próprio dicionário e não representam variáveis.

Após a exclusão desses registros, o inventário final contém **458 variáveis**.

As colunas referentes aos anos indicam a disponibilidade das variáveis para coleta entre 2008 e 2023.

In [ ]:
inventario_censo = inventario_censo[
    inventario_censo["N"].apply(lambda x: str(x).isdigit())
].copy()

print("Quantidade de variáveis identificadas:", len(inventario_censo))

Quantidade de variáveis identificadas: 458


In [ ]:
print("\nDistribuição dos tipos de variáveis:")
print(inventario_censo["Tipo"].value_counts())

print("\nTotal:", inventario_censo["Tipo"].value_counts().sum())


Distribuição dos tipos de variáveis:
Tipo
Num     439
Char     17
Data      2
Name: count, dtype: int64

Total: 458


In [ ]:
print("Variáveis disponíveis no dicionário:\n")

for _, linha in inventario_censo.iterrows():
    print(f"{int(linha['N']):>3} - {linha['Nome da Variável']}")

Variáveis disponíveis no dicionário:

  1 - NU_ANO_CENSO
  2 - NO_REGIAO
  3 - CO_REGIAO
  4 - NO_UF
  5 - SG_UF
  6 - CO_UF
  7 - NO_MUNICIPIO
  8 - CO_MUNICIPIO
  9 - NO_REGIAO_GEOG_INTERM
 10 - CO_REGIAO_GEOG_INTERM
 11 - NO_REGIAO_GEOG_IMED
 12 - CO_REGIAO_GEOG_IMED
 13 - NO_MESORREGIAO
 14 - CO_MESORREGIAO
 15 - NO_MICRORREGIAO
 16 - CO_MICRORREGIAO
 17 - CO_DISTRITO
 18 - NO_ENTIDADE
 19 - CO_ENTIDADE
 20 - TP_DEPENDENCIA
 21 - TP_CATEGORIA_ESCOLA_PRIVADA
 22 - TP_LOCALIZACAO
 23 - TP_LOCALIZACAO_DIFERENCIADA
 24 - DS_ENDERECO
 25 - NU_ENDERECO
 26 - DS_COMPLEMENTO
 27 - NO_BAIRRO
 28 - CO_CEP
 29 - NU_DDD
 30 - NU_TELEFONE
 31 - TP_SITUACAO_FUNCIONAMENTO
 32 - CO_ORGAO_REGIONAL
 33 - DT_ANO_LETIVO_INICIO
 34 - DT_ANO_LETIVO_TERMINO
 35 - IN_VINCULO_SECRETARIA_EDUCACAO
 36 - IN_VINCULO_SEGURANCA_PUBLICA
 37 - IN_VINCULO_SECRETARIA_SAUDE
 38 - IN_VINCULO_OUTRO_ORGAO
 39 - IN_PODER_PUBLICO_PARCERIA
 40 - TP_PODER_PUBLICO_PARCERIA
 41 - IN_CONVENIADA_PP
 42 - TP_CONVENIO_PODER_PUBLI

### 2.3 Análise da estrutura das variáveis

Após a identificação das 458 variáveis, será analisado o padrão de nomenclatura utilizado pelo dicionário do Censo Escolar.

Essa etapa tem como objetivo compreender como as variáveis estão organizadas e quais tipos de informação são disponibilizados, sem realizar ainda a seleção das variáveis que serão utilizadas no projeto

In [ ]:
# Análise dos prefixos utilizados na nomenclatura das variáveis

prefixos = (
    inventario_censo["Nome da Variável"]
    .str.split("_")
    .str[0]
)

print("Quantidade de variáveis por prefixo:\n")
print(prefixos.value_counts())

Quantidade de variáveis por prefixo:

Nome da Variável
IN    239
QT    167
CO     16
TP     16
NO      9
NU      6
DT      2
DS      2
SG      1
Name: count, dtype: int64


Observação sobre a nomenclatura das variáveis

A análise dos nomes das 458 variáveis identificou 9 prefixos distintos:

- `IN`: 239 variáveis
- `QT`: 167 variáveis
- `CO`: 16 variáveis
- `TP`: 16 variáveis
- `NO`: 9 variáveis
- `NU`: 6 variáveis
- `DT`: 2 variáveis
- `DS`: 2 variáveis
- `SG`: 1 variável

Os prefixos `IN` e `QT` concentram a maior parte das variáveis do dicionário, indicando uma predominância de indicadores e informações quantitativas na estrutura analisada.

A classificação foi realizada a partir do padrão de nomenclatura apresentado no próprio nome das variáveis.

In [ ]:
distribuicao_prefixo_tipo = pd.crosstab(
    prefixos,
    inventario_censo["Tipo"]
)

distribuicao_prefixo_tipo

Tipo,Char,Data,Num
Nome da Variável,,,
CO,2,0,14
DS,2,0,0
DT,0,2,0
IN,0,0,239
NO,9,0,0
NU,3,0,3
QT,0,0,167
SG,1,0,0
TP,0,0,16


### 2.4 Padrão de nomenclatura das variáveis

A partir da análise dos nomes e das respectivas descrições apresentadas no dicionário de variáveis do Censo Escolar, foram identificados padrões de nomenclatura associados à natureza das informações.

Os prefixos abaixo foram interpretados com base nos exemplos e descrições presentes no próprio dicionário:

| Prefixo | Interpretação | Exemplo |
|---|---|---|
| `IN` | Indicador | `IN_BIBLIOTECA` |
| `QT` | Quantidade | `QT_SALAS_EXISTENTES` |
| `CO` | Código | `CO_MUNICIPIO` |
| `TP` | Tipo | `TP_DEPENDENCIA` |
| `NO` | Nome | `NO_MUNICIPIO` |
| `NU` | Número | `NU_ANO_CENSO` |
| `DT` | Data | variáveis `DT_` |
| `DS` | Descrição | variáveis `DS_` |
| `SG` | Sigla | `SG_UF` |

> **Observação:** esta classificação representa o padrão de nomenclatura identificado durante a análise do dicionário e não constitui uma alteração ou padronização dos nomes originais das variáveis.

### 2.5 Suplemento de Cursos Técnicos

A segunda estrutura disponibilizada no dicionário do Censo Escolar
2023 corresponde ao `suplemento_cursos_tecnicos`.

Nesta etapa será realizado o levantamento de sua estrutura e das
variáveis disponíveis, seguindo o mesmo procedimento aplicado à
estrutura `microdados_unidade_coleta`.

O objetivo é identificar as informações disponíveis e avaliar sua
possível contribuição para o projeto, sem realizar ainda a seleção
das variáveis que serão utilizadas nas camadas posteriores.

In [ ]:
df_suplemento = pd.read_excel(
    arquivo_dicionario,
    sheet_name="suplemento_cursos_tecnicos",
    header=None
)

print("Linhas:", df_suplemento.shape[0])
print("Colunas:", df_suplemento.shape[1])

Linhas: 47
Colunas: 24


In [ ]:
df_suplemento.head(15)

,0,1,2,3,4,5,6,7,8,9,...,14,15,16,17,18,19,20,21,22,23
0,NaN,Alteração de nomenclatura ou metodologia de cá...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
1,NaN,Variável nova,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
2,NaN,Descontinuidade,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
3,Dicionário de Variáveis - Tabela de Cursos Téc...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
4,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
5,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
6,N,Nome da Variável,Descrição da Variável,Tipo,Tam.(1),Categoria,"Coleta por ano (""s""=sim;""n""=não )",NaN,NaN,NaN,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,Notas Importantes
7,NaN,NaN,NaN,NaN,NaN,NaN,07,08,09,10,...,15,16,17,18,19,20,21,22,23,NaN
8,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
9,1,NU_ANO_CENSO,Ano do Censo,Num,4,NaN,s,s,s,s,...,s,s,s,s,s,s,s,s,s,NaN


In [ ]:
df_suplemento = pd.read_excel(
    arquivo_dicionario,
    sheet_name="suplemento_cursos_tecnicos",
    header=6
)

print("Linhas:", df_suplemento.shape[0])
print("Colunas:", df_suplemento.shape[1])

print("\nColunas:")
for coluna in df_suplemento.columns:
    print(f"- {coluna}")

Linhas: 40
Colunas: 24

Colunas:
- N
- Nome da Variável
- Descrição da Variável
- Tipo
- Tam.(1)
- Categoria
- Coleta por ano ("s"=sim;"n"=não ) 
- Unnamed: 7
- Unnamed: 8
- Unnamed: 9
- Unnamed: 10
- Unnamed: 11
- Unnamed: 12
- Unnamed: 13
- Unnamed: 14
- Unnamed: 15
- Unnamed: 16
- Unnamed: 17
- Unnamed: 18
- Unnamed: 19
- Unnamed: 20
- Unnamed: 21
- Unnamed: 22
- Notas Importantes


In [ ]:
# Identifica os anos de coleta
anos_suplemento = df_suplemento.iloc[0, 7:23].tolist()

print("Anos de coleta:")
print(anos_suplemento)

Anos de coleta:
['08', '09', '10', '11', '12', '13', '14', '15', '16', '17', '18', '19', '20', '21', '22', '23']


In [ ]:
# Renomeia as colunas de anos
novos_nomes_suplemento = (
    list(df_suplemento.columns[:7])
    + anos_suplemento
    + ["Notas Importantes"]
)

df_suplemento.columns = novos_nomes_suplemento

print("Quantidade de colunas:", len(df_suplemento.columns))

print("\nColunas:")
for coluna in df_suplemento.columns:
    print(f"- {coluna}")

Quantidade de colunas: 24

Colunas:
- N
- Nome da Variável
- Descrição da Variável
- Tipo
- Tam.(1)
- Categoria
- Coleta por ano ("s"=sim;"n"=não ) 
- 08
- 09
- 10
- 11
- 12
- 13
- 14
- 15
- 16
- 17
- 18
- 19
- 20
- 21
- 22
- 23
- Notas Importantes


In [ ]:
linhas_sem_n_suplemento = df_suplemento[
    df_suplemento["N"].isna()
]

print(
    "Linhas sem número de variável:",
    len(linhas_sem_n_suplemento)
)

display(linhas_sem_n_suplemento)

Linhas sem número de variável: 3


,N,Nome da Variável,Descrição da Variável,Tipo,Tam.(1),Categoria,"Coleta por ano (""s""=sim;""n""=não )",08,09,10,...,15,16,17,18,19,20,21,22,23,Notas Importantes
0,NaN,NaN,NaN,NaN,NaN,NaN,07,08,09,10,...,15,16,17,18,19,20,21,22,23,NaN
1,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
31,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN


In [ ]:
# Identifica as linhas que correspondem efetivamente às variáveis
inventario_suplemento = df_suplemento[
    df_suplemento["N"].notna()
].copy()

print("Quantidade de variáveis identificadas:", len(inventario_suplemento))

Quantidade de variáveis identificadas: 37


In [ ]:
# Inventário completo das variáveis do suplemento de cursos técnicos

display(
    inventario_suplemento[
        ["N", "Nome da Variável", "Descrição da Variável", "Tipo", "Tam.(1)", "Categoria"]
    ]
)

,N,Nome da Variável,Descrição da Variável,Tipo,Tam.(1),Categoria
2,1,NU_ANO_CENSO,Ano do Censo,Num,4.0,NaN
3,2,NO_REGIAO,Nome da Região Geográfica,Char,20.0,NaN
4,3,CO_REGIAO,Código da Região Geográfica,Num,1.0,NaN
5,4,NO_UF,Nome da Unidade da Federação,Char,50.0,NaN
6,5,SG_UF,Sigla da Unidade da Federação,Char,2.0,NaN
7,6,CO_UF,Código da Unidade da Federação,Num,2.0,NaN
8,7,NO_MUNICIPIO,Nome do Município,Char,150.0,NaN
9,8,CO_MUNICIPIO,Código do Município,Num,7.0,NaN
10,9,TP_LOCALIZACAO,Localização,Num,1.0,1 - Urbana\n2 - Rural
11,10,TP_LOCALIZACAO_DIFERENCIADA,Localização diferenciada da escola,Num,1.0,0 - A escola não está em área de localização d...


In [ ]:
print("Tipos de variáveis:")
print(inventario_suplemento["Tipo"].value_counts())

print("\nTotal:", inventario_suplemento["Tipo"].value_counts().sum())

Tipos de variáveis:
Tipo
Num     22
Char     7
Name: count, dtype: int64

Total: 29


In [ ]:
prefixos_suplemento = (
    inventario_suplemento["Nome da Variável"]
    .str.split("_")
    .str[0]
)

print("Quantidade de variáveis por prefixo:\n")
print(prefixos_suplemento.value_counts())

Quantidade de variáveis por prefixo:

Nome da Variável
QT    12
NO     6
CO     5
TP     3
NU     1
SG     1
ID     1
Name: count, dtype: int64


In [ ]:
# Identifica as variáveis que não possuem Tipo informado

variaveis_sem_tipo = inventario_suplemento[
    inventario_suplemento["Tipo"].isna()
]

print("Variáveis sem tipo informado:", len(variaveis_sem_tipo))

display(
    variaveis_sem_tipo[
        ["N", "Nome da Variável", "Descrição da Variável", "Tipo"]
    ]
)

Variáveis sem tipo informado: 8


,N,Nome da Variável,Descrição da Variável,Tipo
32,Notas:,NaN,NaN,NaN
33,1) Tamanho do campo em termos de caracteres oc...,NaN,NaN,NaN
34,2) O mesmo aluno pode ter mais de uma matrícul...,NaN,NaN,NaN
35,3) Não inclui matrículas e nem cursos em turma...,NaN,NaN,NaN
36,4) A Educação Profissional Técnica de Nível Mé...,NaN,NaN,NaN
37,5) O número de cursos técnicos considera o côm...,NaN,NaN,NaN
38,"6) O Ensino Médio - Normal/Magistério, não con...",NaN,NaN,NaN
39,7) O mesmo curso técnico pode ser ofertado em ...,NaN,NaN,NaN


In [ ]:
# Mantém somente as linhas que correspondem às variáveis
inventario_suplemento = inventario_suplemento[
    pd.to_numeric(inventario_suplemento["N"], errors="coerce").notna()
].copy()

print("Quantidade de variáveis identificadas:", len(inventario_suplemento))

Quantidade de variáveis identificadas: 29


Observação sobre a estrutura do dicionário

A aba `suplemento_cursos_tecnicos` possui 40 linhas e 24 colunas após a leitura do arquivo.

Durante a análise da estrutura, foram identificadas:

- 3 linhas estruturais do próprio dicionário;
- 8 linhas finais contendo notas e observações sobre as variáveis.

Após a análise, verificou-se que **29 linhas correspondem efetivamente às variáveis descritas no dicionário**.

As 29 variáveis apresentam informações relacionadas à oferta e às matrículas em cursos técnicos, incluindo identificação geográfica, características dos cursos e quantidades de cursos e matrículas.

In [ ]:
# Análise dos prefixos das variáveis do suplemento

prefixos_suplemento = (
    inventario_suplemento["Nome da Variável"]
    .str.split("_")
    .str[0]
)

print("Quantidade de variáveis por prefixo:\n")
print(prefixos_suplemento.value_counts())

print("\nTotal:", prefixos_suplemento.value_counts().sum())

Quantidade de variáveis por prefixo:

Nome da Variável
QT    12
NO     6
CO     5
TP     3
NU     1
SG     1
ID     1
Name: count, dtype: int64

Total: 29


Observação sobre a nomenclatura das variáveis

A análise das 29 variáveis do `suplemento_cursos_tecnicos` identificou 7 prefixos distintos:

- `QT`: 12 variáveis
- `NO`: 6 variáveis
- `CO`: 5 variáveis
- `TP`: 3 variáveis
- `NU`: 1 variável
- `SG`: 1 variável
- `ID`: 1 variável

Os prefixos `QT`, `NO` e `CO` concentram a maior parte das variáveis dessa estrutura, refletindo a presença de informações quantitativas, nomes e códigos relacionados aos cursos técnicos.

A classificação foi realizada a partir do padrão de nomenclatura apresentado no próprio dicionário.

In [ ]:
# Lista completa das variáveis do suplemento de cursos técnicos

print("Variáveis disponíveis no suplemento:\n")

for _, linha in inventario_suplemento.iterrows():
    print(f"{int(linha['N']):>2} - {linha['Nome da Variável']}")

Variáveis disponíveis no suplemento:

 1 - NU_ANO_CENSO
 2 - NO_REGIAO
 3 - CO_REGIAO
 4 - NO_UF
 5 - SG_UF
 6 - CO_UF
 7 - NO_MUNICIPIO
 8 - CO_MUNICIPIO
 9 - TP_LOCALIZACAO
10 - TP_LOCALIZACAO_DIFERENCIADA
11 - TP_DEPENDENCIA
12 - NO_ENTIDADE
13 - CO_ENTIDADE
14 - NO_AREA_CURSO_PROFISSIONAL
15 - ID_AREA_CURSO_PROFISSIONAL
16 - NO_CURSO_EDUC_PROFISSIONAL
17 - CO_CURSO_EDUC_PROFISSIONAL
18 - QT_CURSO_TEC
19 - QT_MAT_CURSO_TEC
20 - QT_CURSO_TEC_CT
21 - QT_MAT_CURSO_TEC_CT
22 - QT_CURSO_TEC_NM
23 - QT_MAT_CURSO_TEC_NM
24 - QT_CURSO_TEC_CONC
25 - QT_MAT_CURSO_TEC_CONC
26 - QT_CURSO_TEC_SUBS
27 - QT_MAT_TEC_SUBS
28 - QT_CURSO_TEC_EJA
29 - QT_MAT_TEC_EJA


Observação sobre o suplemento de cursos técnicos

O `suplemento_cursos_tecnicos` possui 29 variáveis.

A estrutura apresenta variáveis de identificação geográfica e da unidade escolar, como ano do Censo, região, UF, município e entidade, além de variáveis específicas relacionadas à educação profissional técnica.

Entre as informações específicas estão a identificação das áreas e cursos técnicos e as quantidades de cursos e matrículas, incluindo diferentes formas de oferta, como Ensino Médio Integrado, concomitante, subsequente e EJA.

Dessa forma, o suplemento complementa as informações da estrutura `microdados_unidade_coleta` com dados específicos sobre a oferta de cursos técnicos.

Padrão de nomenclatura das variáveis do suplemento

A partir da análise dos nomes e das respectivas descrições apresentadas no dicionário de variáveis do Censo Escolar, foram identificados padrões de nomenclatura associados à natureza das informações.

Os prefixos abaixo foram interpretados com base nos exemplos e descrições presentes no próprio dicionário:

| Prefixo | Interpretação | Exemplo |
|---|---|---|
| `IN` | Indicador | `IN_BIBLIOTECA` |
| `QT` | Quantidade | `QT_SALAS_EXISTENTES` |
| `CO` | Código | `CO_MUNICIPIO` |
| `TP` | Tipo | `TP_DEPENDENCIA` |
| `NO` | Nome | `NO_MUNICIPIO` |
| `NU` | Número | `NU_ANO_CENSO` |
| `DT` | Data | Variáveis `DT_` |
| `DS` | Descrição | Variáveis `DS_` |
| `SG` | Sigla | `SG_UF` |
| `ID` | Identificador | `ID_AREA_CURSO_PROFISSIONAL` |

> **Observação:** esta classificação representa o padrão de nomenclatura identificado durante a análise do dicionário e não constitui uma alteração ou padronização dos nomes originais das variáveis.

In [ ]:
print("Tipos de variáveis:")
print(inventario_suplemento["Tipo"].value_counts())

print("\nTotal:", inventario_suplemento["Tipo"].value_counts().sum())

Tipos de variáveis:
Tipo
Num     22
Char     7
Name: count, dtype: int64

Total: 29


In [ ]:
print("Quantidade de variáveis:", len(inventario_suplemento))

Quantidade de variáveis: 29


Síntese da estrutura `suplemento_cursos_tecnicos`

Após a análise da estrutura do dicionário, foram identificadas **29 variáveis** efetivamente descritas.

A distribuição dos tipos de variáveis é:

- **22 variáveis numéricas (`Num`)**
- **7 variáveis de caracteres (`Char`)**

A estrutura apresenta informações de identificação geográfica e das unidades escolares, além de informações específicas sobre a educação profissional técnica, incluindo áreas, cursos, quantidade de cursos e matrículas por modalidade de oferta.

A análise também identificou 7 padrões de nomenclatura (`QT`, `NO`, `CO`, `TP`, `NU`, `SG` e `ID`), utilizados para caracterizar as informações disponíveis no dicionário.

### 2.6 Síntese do Censo Escolar

| Estrutura | Variáveis | Principais informações |
|---|---:|---|
| `microdados_unidade_coleta` | 458 | Características das unidades escolares, localização, infraestrutura, equipamentos, acessibilidade, profissionais e organização escolar. |
| `suplemento_cursos_tecnicos` | 29 | Informações específicas sobre educação profissional técnica, incluindo áreas, cursos, oferta e matrículas. |
| **Total** | **487** | — |

Observação

A análise das duas estruturas resultou em **487 variáveis** documentadas, sendo 458 da estrutura `microdados_unidade_coleta` e 29 do `suplemento_cursos_tecnicos`.

A seleção das variáveis que serão utilizadas nas etapas posteriores não é realizada neste notebook. Este levantamento tem como objetivo documentar as informações disponibilizadas pelas fontes e subsidiar as etapas de definição das análises e das camadas posteriores do projeto.

## Inventário do Censo Escolar 2023

In [ ]:
# Inventário consolidado — Censo Escolar 2023

inventario_censo_geral = pd.concat([
    inventario_censo[["Nome da Variável", "Tipo"]].assign(
        Estrutura="microdados_unidade_coleta"
    ),
    inventario_suplemento[["Nome da Variável", "Tipo"]].assign(
        Estrutura="suplemento_cursos_tecnicos"
    )
], ignore_index=True)

inventario_censo_geral["Tipo"] = inventario_censo_geral["Tipo"].replace({
    "Char": "String"
})

inventario_censo_geral = inventario_censo_geral[
    ["Estrutura", "Nome da Variável", "Tipo"]
]

pd.set_option("display.max_rows", None)

display(inventario_censo_geral)

,Estrutura,Nome da Variável,Tipo
0,microdados_unidade_coleta,NU_ANO_CENSO,Num
1,microdados_unidade_coleta,NO_REGIAO,String
2,microdados_unidade_coleta,CO_REGIAO,Num
3,microdados_unidade_coleta,NO_UF,String
4,microdados_unidade_coleta,SG_UF,String
5,microdados_unidade_coleta,CO_UF,Num
6,microdados_unidade_coleta,NO_MUNICIPIO,String
7,microdados_unidade_coleta,CO_MUNICIPIO,Num
8,microdados_unidade_coleta,NO_REGIAO_GEOG_INTERM,String
9,microdados_unidade_coleta,CO_REGIAO_GEOG_INTERM,Num


# 🏫 Fonte Complementar 2 — INSE 2023 — Indicador de Nível Socioeconômico

O INSE 2023 foi incorporado ao projeto como uma fonte externa complementar, não estando entre as fontes originalmente relacionadas na documentação do trabalho.

A inclusão da fonte foi considerada por sua aderência ao contexto educacional e ao recorte temporal de 2023.

O conjunto disponibilizado apresenta informações do Indicador de Nível Socioeconômico em diferentes níveis de agregação: Brasil, Unidades da Federação, municípios e escolas.

In [ ]:
# Upload dos arquivos do INSE 2023

from google.colab import files
import pandas as pd

arquivos = files.upload()

print("\nArquivos carregados:")
for arquivo in arquivos:
    print("-", arquivo)

Saving INSE_2023_brasil.xlsx to INSE_2023_brasil.xlsx
Saving INSE_2023_escolas.xlsx to INSE_2023_escolas.xlsx
Saving INSE_2023_estados.xlsx to INSE_2023_estados.xlsx
Saving INSE_2023_municipios.xlsx to INSE_2023_municipios.xlsx

Arquivos carregados:
- INSE_2023_brasil.xlsx
- INSE_2023_escolas.xlsx
- INSE_2023_estados.xlsx
- INSE_2023_municipios.xlsx


In [ ]:
# Identificação dos arquivos do INSE 2023

arquivos_inse = {
    "Brasil": "INSE_2023_brasil.xlsx",
    "Estados": "INSE_2023_estados.xlsx",
    "Municípios": "INSE_2023_municipios.xlsx",
    "Escolas": "INSE_2023_escolas.xlsx"
}

print("Arquivos identificados:")
for nivel, arquivo in arquivos_inse.items():
    print(f"- {nivel}: {arquivo}")

Arquivos identificados:
- Brasil: INSE_2023_brasil.xlsx
- Estados: INSE_2023_estados.xlsx
- Municípios: INSE_2023_municipios.xlsx
- Escolas: INSE_2023_escolas.xlsx


In [ ]:
# Levantamento inicial da estrutura dos arquivos do INSE 2023

for nivel, arquivo in arquivos_inse.items():

    xls = pd.ExcelFile(arquivo)

    print("\n" + "=" * 70)
    print(f"{nivel.upper()}")
    print("=" * 70)

    print("Abas disponíveis:", xls.sheet_names)

    for aba in xls.sheet_names:
        df_temp = pd.read_excel(arquivo, sheet_name=aba)

        print(
            f"  {aba}: "
            f"{df_temp.shape[0]:,} linhas × "
            f"{df_temp.shape[1]} colunas"
        )


BRASIL
Abas disponíveis: ['INSE_BR_2023', 'Dicionário']
  INSE_BR_2023: 63 linhas × 14 colunas
  Dicionário: 14 linhas × 2 colunas

ESTADOS
Abas disponíveis: ['INSE_UF_2023', 'Dicionário']
  INSE_UF_2023: 1,552 linhas × 17 colunas
  Dicionário: 17 linhas × 2 colunas

MUNICÍPIOS
Abas disponíveis: ['INSE_MUN_2023', 'Dicionário']
  INSE_MUN_2023: 125,741 linhas × 19 colunas
  Dicionário: 19 linhas × 2 colunas

ESCOLAS
Abas disponíveis: ['INSE_ESC_2023', 'Dicionário']
  INSE_ESC_2023: 69,756 linhas × 22 colunas
  Dicionário: 22 linhas × 2 colunas


In [ ]:
# Listagem das variáveis disponíveis nas tabelas do INSE 2023

for nivel, arquivo in arquivos_inse.items():

    xls = pd.ExcelFile(arquivo)

    # Primeira aba = tabela principal
    aba_principal = xls.sheet_names[0]

    df_temp = pd.read_excel(
        arquivo,
        sheet_name=aba_principal
    )

    print("\n" + "=" * 70)
    print(f"{nivel.upper()} - {aba_principal}")
    print("=" * 70)

    print(f"Quantidade de variáveis: {len(df_temp.columns)}")
    print("\nColunas:")

    for coluna in df_temp.columns:
        print(f"- {coluna}")


BRASIL - INSE_BR_2023
Quantidade de variáveis: 14

Colunas:
- NU_ANO_SAEB
- TP_TIPO_REDE
- TP_LOCALIZACAO
- TP_CAPITAL
- QTD_ALUNOS_INSE
- MEDIA_INSE
- PC_NIVEL_1
- PC_NIVEL_2
- PC_NIVEL_3
- PC_NIVEL_4
- PC_NIVEL_5
- PC_NIVEL_6
- PC_NIVEL_7
- PC_NIVEL_8

ESTADOS - INSE_UF_2023
Quantidade de variáveis: 17

Colunas:
- NU_ANO_SAEB
- CO_UF
- SG_UF
- NO_UF
- TP_TIPO_REDE
- TP_LOCALIZACAO
- TP_CAPITAL
- QTD_ALUNOS_INSE
- MEDIA_INSE
- PC_NIVEL_1
- PC_NIVEL_2
- PC_NIVEL_3
- PC_NIVEL_4
- PC_NIVEL_5
- PC_NIVEL_6
- PC_NIVEL_7
- PC_NIVEL_8

MUNICÍPIOS - INSE_MUN_2023
Quantidade de variáveis: 19

Colunas:
- NU_ANO_SAEB
- CO_UF
- SG_UF
- NO_UF
- CO_MUNICIPIO
- NO_MUNICIPIO
- TP_TIPO_REDE
- TP_LOCALIZACAO
- TP_CAPITAL
- QTD_ALUNOS_INSE
- MEDIA_INSE
- PC_NIVEL_1
- PC_NIVEL_2
- PC_NIVEL_3
- PC_NIVEL_4
- PC_NIVEL_5
- PC_NIVEL_6
- PC_NIVEL_7
- PC_NIVEL_8

ESCOLAS - INSE_ESC_2023
Quantidade de variáveis: 22

Colunas:
- NU_ANO_SAEB
- CO_UF
- SG_UF
- NO_UF
- CO_MUNICIPIO
- NO_MUNICIPIO
- ID_ESCOLA
- NO_ESC

In [ ]:
# Leitura dos dicionários das bases do INSE 2023

for nivel, arquivo in arquivos_inse.items():

    dicionario = pd.read_excel(
        arquivo,
        sheet_name="Dicionário"
    )

    print("\n" + "=" * 70)
    print(f"DICIONÁRIO - {nivel.upper()}")
    print("=" * 70)

    print(f"Quantidade de registros: {len(dicionario)}")
    print("\nColunas do dicionário:")

    for coluna in dicionario.columns:
        print(f"- {coluna}")

    display(dicionario)


DICIONÁRIO - BRASIL
Quantidade de registros: 14

Colunas do dicionário:
- Nome
- Descrição


,Nome,Descrição
0,NU_ANO_SAEB,Ano de aplicação do Saeb
1,TP_TIPO_REDE,Dependência Administrativa:\n0 - Total (Federa...
2,TP_LOCALIZACAO,Localização:\n0 - Total (Urbana e Rural)\n1 - ...
3,TP_CAPITAL,Capital / Interior:\n0 - Total (Capital e Inte...
4,QTD_ALUNOS_INSE,Quantidade de Alunos com INSE calculado utiliz...
5,MEDIA_INSE,Média ponderada do Indicador de Nível Socioeco...
6,PC_NIVEL_1,Percentual de alunos da Escola classificados n...
7,PC_NIVEL_2,Percentual de alunos da Escola classificados n...
8,PC_NIVEL_3,Percentual de alunos da Escola classificados n...
9,PC_NIVEL_4,Percentual de alunos da Escola classificados n...



DICIONÁRIO - ESTADOS
Quantidade de registros: 17

Colunas do dicionário:
- Nome
- Descrição


,Nome,Descrição
0,NU_ANO_SAEB,Ano de aplicação do Saeb
1,CO_UF,Código da Unidade da Federação
2,SG_UF,Sigla da Unidade da Federação
3,NO_UF,Nome da Unidade da Federação
4,TP_TIPO_REDE,Dependência Administrativa:\n0 - Total (Federa...
5,TP_LOCALIZACAO,Localização:\n0 - Total (Urbana e Rural)\n1 - ...
6,TP_CAPITAL,Capital / Interior:\n0 - Total (Capital e Inte...
7,QTD_ALUNOS_INSE,Quantidade de Alunos com INSE calculado utiliz...
8,MEDIA_INSE,Média ponderada do Indicador de Nível Socioeco...
9,PC_NIVEL_1,Percentual de alunos do estrato da UF classifi...



DICIONÁRIO - MUNICÍPIOS
Quantidade de registros: 19

Colunas do dicionário:
- Nome
- Descrição


,Nome,Descrição
0,NU_ANO_SAEB,Ano de aplicação do Saeb
1,CO_UF,Código da Unidade da Federação
2,SG_UF,Sigla da Unidade da Federação
3,NO_UF,Nome da Unidade da Federação
4,CO_MUNICIPIO,Código do Município
5,NO_MUNICIPIO,Nome do Município
6,TP_TIPO_REDE,Dependência Administrativa:\n1 - Federal\n2 - ...
7,TP_LOCALIZACAO,Localização:\n0 - Total (Urbana e Rural)\n1 - ...
8,TP_CAPITAL,Área da Escola (relacionado ao Município):\n1 ...
9,QTD_ALUNOS_INSE,Quantidade de Alunos com INSE calculado utiliz...



DICIONÁRIO - ESCOLAS
Quantidade de registros: 22

Colunas do dicionário:
- Nome
- Descrição


,Nome,Descrição
0,NU_ANO_SAEB,Ano de aplicação do Saeb
1,CO_UF,Código da Unidade da Federação
2,SG_UF,Sigla da Unidade da Federação
3,NO_UF,Nome da Unidade da Federação
4,CO_MUNICIPIO,Código do Município
5,NO_MUNICIPIO,Nome do Município
6,ID_ESCOLA,Código da Escola no CENSO Escolar
7,NO_ESCOLA,Nome da Escola no CENSO Escolar
8,TP_TIPO_REDE,Dependência Administrativa da Escola:\n1 - Fed...
9,TP_LOCALIZACAO,Localização da Escola:\n1 - Urbana\n2 - Rural


In [ ]:
# Tipos de dados das variáveis do INSE 2023

for nivel, arquivo in arquivos_inse.items():

    xls = pd.ExcelFile(arquivo)
    aba_principal = xls.sheet_names[0]

    df_temp = pd.read_excel(
        arquivo,
        sheet_name=aba_principal
    )

    print("\n" + "=" * 70)
    print(f"{nivel.upper()} - TIPOS DE DADOS")
    print("=" * 70)

    print(df_temp.dtypes)


BRASIL - TIPOS DE DADOS
NU_ANO_SAEB          int64
TP_TIPO_REDE         int64
TP_LOCALIZACAO       int64
TP_CAPITAL           int64
QTD_ALUNOS_INSE      int64
MEDIA_INSE         float64
PC_NIVEL_1         float64
PC_NIVEL_2         float64
PC_NIVEL_3         float64
PC_NIVEL_4         float64
PC_NIVEL_5         float64
PC_NIVEL_6         float64
PC_NIVEL_7         float64
PC_NIVEL_8         float64
dtype: object

ESTADOS - TIPOS DE DADOS
NU_ANO_SAEB          int64
CO_UF                int64
SG_UF               object
NO_UF               object
TP_TIPO_REDE         int64
TP_LOCALIZACAO       int64
TP_CAPITAL           int64
QTD_ALUNOS_INSE      int64
MEDIA_INSE         float64
PC_NIVEL_1         float64
PC_NIVEL_2         float64
PC_NIVEL_3         float64
PC_NIVEL_4         float64
PC_NIVEL_5         float64
PC_NIVEL_6         float64
PC_NIVEL_7         float64
PC_NIVEL_8         float64
dtype: object

MUNICÍPIOS - TIPOS DE DADOS
NU_ANO_SAEB          int64
CO_UF                int64
S

In [ ]:
# Verificação de valores nulos nas tabelas do INSE 2023

for nivel, arquivo in arquivos_inse.items():

    xls = pd.ExcelFile(arquivo)
    aba_principal = xls.sheet_names[0]

    df_temp = pd.read_excel(
        arquivo,
        sheet_name=aba_principal
    )

    nulos = df_temp.isna().sum()

    print("\n" + "=" * 70)
    print(f"{nivel.upper()} - VALORES NULOS")
    print("=" * 70)

    print(nulos[nulos > 0])


BRASIL - VALORES NULOS
Series([], dtype: int64)

ESTADOS - VALORES NULOS
Series([], dtype: int64)

MUNICÍPIOS - VALORES NULOS
MEDIA_INSE      890
PC_NIVEL_1       85
PC_NIVEL_2     4460
PC_NIVEL_3     1612
PC_NIVEL_4      520
PC_NIVEL_5      774
PC_NIVEL_6     2950
PC_NIVEL_7     7811
PC_NIVEL_8    42414
dtype: int64

ESCOLAS - VALORES NULOS
PC_NIVEL_1      265
PC_NIVEL_2     6720
PC_NIVEL_3     1922
PC_NIVEL_4     1339
PC_NIVEL_5     3483
PC_NIVEL_6     9030
PC_NIVEL_7    16497
PC_NIVEL_8    46332
dtype: int64


In [ ]:
# Inventário consolidado das bases do INSE 2023

resumo_inse = []

for nivel, arquivo in arquivos_inse.items():

    xls = pd.ExcelFile(arquivo)
    aba_principal = xls.sheet_names[0]

    df_temp = pd.read_excel(
        arquivo,
        sheet_name=aba_principal
    )

    resumo_inse.append({
        "Nível de agregação": nivel,
        "Tabela": aba_principal,
        "Linhas": df_temp.shape[0],
        "Colunas": df_temp.shape[1]
    })

inventario_inse = pd.DataFrame(resumo_inse)

display(inventario_inse)

,Nível de agregação,Tabela,Linhas,Colunas
0,Brasil,INSE_BR_2023,63,14
1,Estados,INSE_UF_2023,1552,17
2,Municípios,INSE_MUN_2023,125741,19
3,Escolas,INSE_ESC_2023,69756,22


## Inventário do INSE 2023

A análise estrutural do INSE 2023 identificou quatro tabelas principais, organizadas em diferentes níveis de agregação: Brasil, Unidades da Federação, municípios e escolas.

| Nível de agregação | Tabela | Registros | Variáveis |
|---|---|---:|---:|
| Brasil | INSE_BR_2023 | 63 | 14 |
| Estados | INSE_UF_2023 | 1.552 | 17 |
| Municípios | INSE_MUN_2023 | 125.741 | 19 |
| Escolas | INSE_ESC_2023 | 69.756 | 22 |

As tabelas apresentam uma estrutura hierárquica de identificação. Conforme aumenta o nível de detalhamento, são incorporadas informações geográficas e de identificação das unidades.

No nível municipal estão disponíveis os códigos e nomes dos municípios (`CO_MUNICIPIO` e `NO_MUNICIPIO`). No nível escolar, além da identificação municipal, estão disponíveis o código e o nome da escola (`ID_ESCOLA` e `NO_ESCOLA`).

As informações centrais do INSE são representadas pela quantidade de alunos considerados no cálculo (`QTD_ALUNOS_INSE`), pela média do indicador (`MEDIA_INSE`) e pela distribuição percentual dos alunos entre os níveis 1 a 8 (`PC_NIVEL_1` a `PC_NIVEL_8`).

Na tabela de escolas também está disponível a variável `INSE_CLASSIFICACAO`, relacionada à classificação do Indicador de Nível Socioeconômico.

A análise dos tipos de dados identificou variáveis numéricas inteiras para os campos de identificação e classificação e variáveis do tipo `float` para a média do INSE e os percentuais de distribuição por nível.

Quanto aos valores ausentes, não foram identificados valores nulos nas tabelas de Brasil e Estados. Na tabela de municípios foram identificados valores ausentes em `MEDIA_INSE` e nas variáveis `PC_NIVEL_1` a `PC_NIVEL_8`. Na tabela de escolas foram identificados valores ausentes nas variáveis `PC_NIVEL_1` a `PC_NIVEL_8`.

Os valores ausentes foram apenas identificados nesta etapa e não receberam tratamento, pois as regras de limpeza e transformação serão definidas posteriormente na camada Prata.

## Considerações sobre a utilização do INSE 2023

O INSE 2023 foi incorporado ao projeto como uma fonte externa complementar. Essa fonte não estava entre as bases originalmente relacionadas na documentação do trabalho.

A inclusão foi definida por sua aderência ao contexto educacional e ao recorte temporal de 2023.

A base apresenta informações em diferentes níveis de agregação, permitindo análises em nível nacional, estadual, municipal e escolar. A definição das tabelas e variáveis que serão efetivamente utilizadas nos cruzamentos será realizada posteriormente, a partir das perguntas analíticas e da modelagem do projeto.

Nesta etapa, o objetivo foi exclusivamente identificar, documentar e avaliar a estrutura da fonte. O tratamento dos dados, a seleção definitiva das variáveis e os relacionamentos entre as bases serão definidos nas etapas posteriores.